OpenRA.Mods.AS/Activities/EnterTeleportNetwork.cs

```diff
@@ -0,0 +1,112 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2015- OpenRA.Mods.AS Developers (see AUTHORS)
+ * This file is a part of a third-party plugin for OpenRA, which is
+ * free software. It is made available to you under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation. For more information, see COPYING.
+ */
+#endregion
+
+using System.Drawing;
+using System.Linq;
+using OpenRA.Mods.AS.Traits;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.AS.Activities
+{
+	class EnterTeleportNetwork : Enter
+	{
+		public EnterTeleportNetwork(Actor self, Actor target, EnterBehaviour enterBehaviour)
+			: base(self, target, enterBehaviour)
+		{ }
+
+		protected override bool CanReserve(Actor self)
+		{
+			// Primary building is where you come out!
+			return !Target.Actor.IsPrimaryTeleportNetworkExit();
+		}
+
+		protected override void OnInside(Actor self)
+		{
+			// entered the teleport network canal but the entrance is dead immediately.
+			if (Target.Actor.IsDead || self.IsDead)
+				return;
+
+			var targettrait = Target.Actor.TraitOrDefault<TeleportNetwork>();
```

Comment:
This is definitely wrong. You need to pass the type here atleast (maybe through the constructor).